File: boinc/test/test_masterurl_failure.php
#!/usr/local/bin/php -q
<?php {
    // $Id$

    $use_proxy_html = 1;

    include_once("test_uc.inc");
    test_msg("master url failure");

    $project = new ProjectUC;

    start_proxy('exit 1 if $nconnections < 4; if_done_kill(); if_done_ping();');

    // TODO: verify it took ??? seconds

    // TODO: time out after ??? seconds and fail this test

    $project->start_servers_and_host();
    $project->validate_all_and_stop();

    test_done();
} ?>

File: test/test_masterurl_failure.py
#!/usr/bin/env python

## $Id$

from test_uc import *

if __name__ == '__main__':
    test_msg("scheduler exponential backoff (master url failure)")
    Proxy('close_connection if $nconnections < 2; ', html=1)
    # Proxy(('print "hi nc=$nconnections start=$start nchars=$nchars url=$url\n"; ' +
    #        'close_connection if $nconnections < 2; '),
    #       html=1)
    # Proxy(( 'print "hi nc=$nconnections start=$start nchars=$nchars url=$url\n"; ' +
    #        ''),
    #       html=1)
    ProjectUC(short_name='test_masterurl_failure')
    run_check_all()

    ## TODO: verify it took ??? seconds
